### PR TITLE
fix(metadata): update cascade options for comic metadata relationship to include removal and orphan handling

### PR DESCRIPTION
**Pull request by balazs-szucs**
## 📝 Description

Possible fix for: #3000 

**Linked Issue:** Fixes #

> **Required.** Every PR must reference an approved issue. If no issue exists, [open one](https://github.com/booklore-app/booklore/issues/new) and wait for maintainer approval before submitting a PR. Unsolicited PRs without a linked issue will be closed.

## 🏷️ Type of Change

- [X] Bug fix
- [ ] New feature
- [ ] Enhancement to existing feature
- [ ] Refactor (no behavior change)
- [ ] Breaking change (existing functionality affected)
- [ ] Documentation update

## 🔧 Changes

-

## 🧪 Testing (MANDATORY)

> **PRs without this section filled out will be closed.** "Tests pass" or "Tested locally" is not sufficient. You must provide specifics.

**Manual testing steps you performed:**

1.
2.
3.

**Regression testing:**

-

**Edge cases covered:**

-

**Test output:**

<details>
<summary>Backend test output (<code>./gradlew test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

<details>
<summary>Frontend test output (<code>ng test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

## 📸 Screen Recording / Screenshots (MANDATORY)

> Every PR must include a **screen recording or screenshots** showing the change working end-to-end in a running local instance (both backend and frontend). This means you must have actually built, run, and tested the code yourself. PRs without visual proof will be closed without review.

---

## ✅ Pre-Submission Checklist

> **All boxes must be checked before requesting review.** Incomplete PRs will be closed without review. No exceptions.

- [ ] This PR is linked to an approved issue
- [ ] Code follows project style guidelines and conventions
- [ ] Branch is up to date with `develop` (merge conflicts resolved)
- [ ] I ran the full stack locally (backend + frontend + database) and verified the change works
- [ ] Automated tests added or updated to cover changes (backend **and** frontend)
- [ ] All tests pass locally and output is pasted above
- [ ] Screen recording or screenshots are attached above proving the change works
- [ ] PR is a single focused change (one bug fix OR one feature, not multiple unrelated changes)
- [ ] PR is reasonably scoped (PRs over 1000+ changed lines will be closed, split into smaller PRs)
- [ ] No unsolicited refactors, cleanups, or "improvements" are bundled in
- [ ] Flyway migration versioning is correct _(if schema was modified)_
- [ ] Documentation PR submitted to [booklore-docs](https://github.com/booklore-app/booklore-docs) _(if user-facing changes)_

### 🤖 AI-Assisted Contributions

> **If any part of this PR was generated or assisted by AI tools (Copilot, Claude, ChatGPT, etc.), all items below are mandatory.** You are fully responsible for every line you submit. "The AI wrote it" is not an excuse, and AI-generated PRs that clearly haven't been reviewed are the #1 reason PRs get closed.

- [ ] I have read and understand every line of this PR and can explain any part of it during review
- [ ] I personally ran the code and verified it works (not just trusted the AI's output)
- [ ] PR is scoped to a single logical change, not a dump of everything the AI suggested
- [ ] Tests validate actual behavior, not just coverage (AI-generated tests often assert nothing meaningful)
- [ ] No dead code, placeholder comments, `TODO`s, or unused scaffolding left behind by AI
- [ ] I did not submit refactors, style changes, or "improvements" the AI suggested beyond the scope of the issue

---

### CODE REVIEW

**Comment by balazs-szucs** [comment]
Not to repeat myself, but eventually we have to get to #2670 because hibernate situation is not maintainable. We need consistency...

If you indicate when I should get the PR ready, I'll do it.

**Comment by balazs-szucs** [comment]
As for this, it's a pure guess.

File: booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java [hunk 1]
```diff
@@ -331,7 +331,7 @@ public void updateSearchText() {
     @JsonIgnore
     private BookEntity book;
 
-    @OneToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE}, fetch = FetchType.LAZY)
+    @OneToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE}, orphanRemoval = true, fetch = FetchType.LAZY)
```
Comment (acx10) [inline booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: hey, thanks for looking into this! the change itself looks reasonable, `CascadeType.REMOVE` + `orphanRemoval = true` makes sense here since `BookEntity` already cascades ALL + orphanRemoval down to `BookMetadataEntity` (line 32 of BookEntity), but that cascade was stopping at `BookMetadataEntity` and not reaching the `ComicMetadataEntity` child. so deleting a book would leave orphaned comic metadata rows behind, which matches the error in #3000.

one thing though, the PR description says "possible fix" so it sounds like this is a guess. were you able to actually reproduce the issue and test the fix? the issue reporter mentioned it only happens on specific entries in their instance, so it'd be good to know if you confirmed the cascade chain works end-to-end (delete a book that has comic metadata and verify the comic_metadata row gets cleaned up too). if you haven't been able to reproduce it, totally understandable given the setup, just want to make sure we're not merging blind.

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Nope. I wasn't, essentially I looked "What's different here" that may cause problems and made educated guess there.

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: I tested this, it worked, but (for me) it worked before also perfectly so that does not mean anything

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Though, we can try to give to the people a test container see where it goes? What do you think about that 🤔

Comment (acx10) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Yeah, makes sense to have the folks who reported the bug test the container first. They'd be the best ones to verify the fix.

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Sounds good, I think containers only built on your PRs so I think you'll need to raise these changes, if yes, feel free to merge that. Or am I mistaken about that?

Comment (acx10) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Do you have a pipeline set up in your fork? We could potentially use that to build the container and share it with the users who reported the issue.

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: I'm afraid, it's broken since Java 25. I haven't bothered to update it. So, no. Worst case scenerio, I can just build it manually and push it to docker hub that way 😅

Comment (DanielBrierton) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: I should be able to build this myself and run it locally to test it out if you don't get a build pushed.

I'm assuming if I just checkout this PR, run `docker build` and tag it, that's all it takes? Or is there more to the build than that? I'll hopefully get properly set up for development at some point soon.

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Sure, I think there are some useful tips for you at: https://github.com/booklore-app/booklore/blob/develop/CONTRIBUTING.md

But feel free to ask, if something not clear.

Comment (DanielBrierton) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Got a chance to build the image there and test it out. Unfortunately it hasn’t resolved the issue. The error I’m now getting is

> Batch update returned unexpected row count from update 0 (expected row count 1 but was 0) [delete from comic_metadata where book_id=?] for entity [org.booklore.model.entity.ComicMetadataEntity with id '5737']

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: This gave me an idea. 1 sec!

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Can you retry with this commit included: [383de22](https://github.com/booklore-app/booklore/pull/3023/commits/383de22e5c9ae614cc87f9458eb4eea04cd637c3)

Comment (DanielBrierton) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Sorry, I missed your response yesterday.

Looks like @acx10 might have fixed this in https://github.com/booklore-app/booklore/pull/3067 already, but I've yet to try it myself. I'll try out their fix first since it's merged, and if that doesn't work I'll come back and try with that commit. Thanks!

Comment (balazs-szucs) [reply booklore-api/src/main/java/org/booklore/model/entity/BookMetadataEntity.java]: Sounds good for me! Thanks guys.
